src/react-apps/applications/ux-editor/src/utils/arrayHelpers/arrayLogic.ts

```diff
@@ -0,0 +1,110 @@
+import { createSelector } from 'reselect';
+
+export interface IUpdateArrayObj {
+  array: any[];
+  object: any;
+}
+
+export interface ITwoArraysObj {
+  array: any[];
+  order: any [];
+}
+
+const pushOrPop = (obj: IUpdateArrayObj) => {
+  const tempObj = obj.array.find((o: any) => o.id === obj.object.id);
+  if (!tempObj) {
+    obj.array.push(obj.object);
+    obj.array.sort(compareOrderNum);
+  } else {
+    const index = obj.array.indexOf(tempObj);
+    obj.array.splice(index, 1);
+  }
+  if (obj.array.length > 0) {
+    markFirstAndLastObject(obj.array);
+  }
+  const cloneOfObj = JSON.parse(JSON.stringify(obj));
+  return cloneOfObj;
+};
+
+const changeOrderNum = (obj: ITwoArraysObj) => {
+  obj.array.forEach((component: any) => {
+    if (obj.order.indexOf(component.id) >= 0) {
+      component.order = obj.order.indexOf(component.id);
+    }
+  });
+  const cloneOfObj = JSON.parse(JSON.stringify(obj));
+  return cloneOfObj;
+};
+
+const compareOrderNum = (a: any, b: any) => {
+  if (a.order < b.order) {
+    return -1;
+  } else if (a.order > b.order) {
+    return 1;
+  } else if (a.order === b.order) {
+    b.order = a.order + 1;
+  }
+  return 0;
+};
+
+const findMissing = (a: any[]) => {
+  const maximum = Math.max.apply(Math, a.map((o) => {
+    return o.order;
+  }));
+  const minimum = Math.min.apply(Math, a.map((o) => {
+    return o.order;
+  }));
+  let index = 0;
+  Array.from(Array(maximum).keys()).forEach((i: number) => {
+    const obj = a.find((x: any) => x.order === i);
+    if (obj) {
+      index = a.indexOf(obj);
+    }
+    if (!obj && i > minimum) {
+      a[index].lastInActiveList = true;
+      a[index + 1].firstInActiveList = true;
+    }
+  });
+  return a;
+};
+
+const markFirstAndLastObject = (array: any[]) => {
+  for (let i = 0; i <= array.length - 1; i++) {
+    if (i === 0 && array.length === 1) {
+      array[i].firstInActiveList = true;
+      array[i].lastInActiveList = true;
+    } else if (i === 0 && array.length > 1) {
+      array[i].firstInActiveList = true;
+      array[i].lastInActiveList = false;
+    } else if (i === array.length - 1) {
+      array[i].firstInActiveList = false;
+      array[i].lastInActiveList = true;
+    } else {
+      array[i].firstInActiveList = false;
+      array[i].lastInActiveList = false;
+    }
+  }
```

Comment:
Couldn't this be as simple as:
```
array[0].firstInActiveList = true;
array[0].lastInActiveList = false;
array[array.length - 1].lastInActiveList = true; 
```
?

Reply:
No, because the ActiveList could include an order like 1, 3, 4. Which means ActiveList[0] should have both firstInActiveList and lastInActiveList set to true in order to get the first and last css class.
The names might be a bit misleading tho..

Reply:
👍